### PR TITLE
Enable Snapper-Rollback on HDD special Image.

### PR DESCRIPTION
**Pull request by MalloZup**

### CODE REVIEW

**Comment by MalloZup** [comment]
How to enable this test:

-  Create a test on prod.
- Create an image with snapshot after upgrade.

**Comment by sysrich** [comment]
Do you have a link to it running on your local systems so we can get a feel for how it's working?

Doesn't this PR prevent testing of poo#9570? https://progress.opensuse.org/issues/9570

There are two discrete scenarios we want to test

We want to test whether or not the boot to snapshot feature works, https://progress.opensuse.org/issues/9570

This test needs to be run on every build, regardless of migration status

and we want to test that migration rollback works, https://progress.opensuse.org/issues/9580

Unless I misunderstand something, this PR seems to address 9580, but blocks 9570

And I don't like that, because that will mean that we will have no coverage of "boot to snapshot" unless migration testing is green

And migration testing might not always be green.

**Comment by MalloZup** [comment]
ok fixing

**Comment by MalloZup** [comment]
OK. boot_into_snapshot test remain into /installation tests.

Snapper_rollback is on /boot  tests and integrated a basic test from boot_into_snapshot( not the whole)

**Comment by MalloZup** [comment]
prrof

**Comment by MalloZup** [comment]
gogo

**Comment by MalloZup** [comment]
Hi guys, what do you think about this PR ? :) 
summarizing:
* we have a new test snapper_rollback in /boot.
* boot_into_snapshot test remain into /installation tests.
* after the merge prod is safe. We have to enable a new tests, with a new-image from a snapshot image.

File: tests/boot/grub_test_snapshot.pm [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        # in upgrade/migration scenario, we want to boot from snapshot 1 before migration.
+        if (get_var("UPGRADE")) {
+            send_key 'down';
+            save_screenshot;
+        }
+        send_key 'ret';
+        # bsc#956046  check if we are in first menu-entry, or not
+        if (check_screen("boot-menu-snapshot-bootmenu")) {
+            send_key 'down', 1;
+            save_screenshot;
```
Comment (sysrich) [inline tests/boot/grub_test_snapshot.pm]: this is a workaround for bug, so should have `record_soft_failure;`

File: tests/boot/grub_test_snapshot.pm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        # in upgrade/migration scenario, we want to boot from snapshot 1 before migration.
+        if (get_var("UPGRADE")) {
+            send_key 'down';
+            save_screenshot;
+        }
+        send_key 'ret';
+        # bsc#956046  check if we are in first menu-entry, or not
+        if (check_screen("boot-menu-snapshot-bootmenu")) {
+            record_soft_failure 'bsc#956046';
+            send_key 'down', 1;
```
Comment (MalloZup) [inline tests/boot/grub_test_snapshot.pm]: updated

File: tests/installation/grub_test.pm [hunk 3]
```diff
@@ -29,6 +29,7 @@ sub run() {
         send_key 'ret';
         # bsc#956046  check if we are in first menu-entry, or not
         if (check_screen("boot-menu-snapshot-bootmenu")) {
+            record_soft_failure 'bsc#956046';
```
Comment (MalloZup) [inline tests/installation/grub_test.pm]: updated

**Comment by sysrich** [comment]
Preparing the following changes to the following scenarios

migration_offline_sle12
+ PUBLISH_HDD_1=SLES-%VERSION%-%ARCH%-default-migration-from-sle12.qcow2
migration_offline_sle12sp1
+ PUBLISH_HDD_1=SLES-%VERSION%-%ARCH%-default-migration-from-sle12sp1.qcow2
migration_offline_sle12_ppc
+ PUBLISH_HDD_1=SLES-%VERSION%-%ARCH%-default-migration-from-sle12.qcow2
migration_offline_sle12sp1_ppc
+ PUBLISH_HDD_1=SLES-%VERSION%-%ARCH%-default-migration-from-sle12sp1.qcow2

Planning the following additional scenarios

rollback_migration_offline_sle12sp1
BOOT_TO_SNAPSHOT=1
UPGRADE=1
START_AFTER_TEST=migration_offline_sle12sp1
HDD_1=SLES-%VERSION%-%ARCH%-default-migration-from-sle12sp1.qcow2

rollback_migration_offline_sle12
BOOT_TO_SNAPSHOT=1
UPGRADE=1
START_AFTER_TEST=migration_offline_sle12
HDD_1=SLES-%VERSION%-%ARCH%-default-migration-from-sle12.qcow2

rollback_migration_offline_sle12_ppc
BOOT_TO_SNAPSHOT=1
UPGRADE=1
START_AFTER_TEST=migration_offline_sle12_ppc
HDD_1=SLES-%VERSION%-%ARCH%-default-migration-from-sle12.qcow2

rollback_migration_offline_sle12sp1_ppc
BOOT_TO_SNAPSHOT=1
UPGRADE=1
START_AFTER_TEST=migration_offline_sle12sp1_ppc
HDD_1=SLES-%VERSION%-%ARCH%-default-migration-from-sle12sp1.qcow2

**Comment by MalloZup** [comment]
ok looks nice

**Comment by sysrich** [comment]
It's not nice - I missed the build number in all the HDDs :)

**Comment by MalloZup** [comment]
ok i mean, i focused on the structure of that .. :)

**Comment by sysrich** [comment]
Scenarios setup and in the Acceptance Job group..lets see how bad it works :)

**Comment by MalloZup** [comment]
Please change the  setup for the test 
* ISO 	SLE-12-SP2-Server-DVD-x86_64-Build1466-Media1.iso -> Remove.
*  set BOOT_HDD_IMAGE :1
* and related variables that concern iso.

### Should look like this. (image name are not of course correct in sync with Prod)

ARCH 	x86_64
BETA 	1
BETA_SDK 	1
BETA_WE 	1
BOOT_HDD_IMAGE 	1
BOOT_TO_SNAPSHOT 	1
BUILD 	1415
BUILD_HA 	0243
BUILD_HA_GEO 	0069
BUILD_SDK 	0626
BUILD_WE 	0193
DESKTOP 	gnome
DISTRI 	sle
DVD 	1
FAKE_SCC_EMAIL 	openqa@suse.de
FAKE_SCC_REGCODE 	SLE-12-SP2-Server-DVD-x86_64-5UH8qgmU6TGmH1nP
FAKE_SCC_REGCODE_HA 	SLE-12-SP2-Server-DVD-x86_64-5UH8qgmU6TGmH1nP
FAKE_SCC_REGCODE_HA_GEO 	SLE-12-SP2-Server-DVD-x86_64-5UH8qgmU6TGmH1nP
FAKE_SCC_REGCODE_WE 	SLE-12-SP2-Server-DVD-x86_64-5UH8qgmU6TGmH1nP
FAKE_SCC_URL 	http://scc.openqa.suse.de
FLAVOR 	Server-DVD
GNOME 	1
HDDFORMAT 	qcow2
HDDSIZEGB 	20
HDD_1 	SLES-12-SP1-GM-allpatterns-x86_64.img
HOST 	localhost
MACHINE 	64bit
NAME 	00000876-sle-12-SP2-Server-DVD-x86_64-Build1415-BOOT_SNAP_NEW
NOAUTOLOGIN 	1
NOIMAGES 	1
NUMDISKS 	1
PACKAGETOINSTALL 	x3270
PATTERNS 	all
REPO_0 	SLE-12-SP2-Server-DVD-x86_64-Build1415-Media1
REPO_10 	SLE-12-SP2-HA-POOL-x86_64-Build0243-Media1
REPO_11 	SLE-12-SP2-HA-POOL-x86_64-Build0243-Media1.license
REPO_12 	SLE-12-SP2-HA-GEO-POOL-s390x-x86_64-Build0069-Media1
REPO_13 	SLE-12-SP2-HA-GEO-POOL-s390x-x86_64-Build0069-Media1.license
REPO_14 	SLE-12-SP2-WE-POOL-x86_64-Build0193-Media1
REPO_15 	SLE-12-SP2-WE-POOL-x86_64-Build0193-Media1.license
REPO_7 	SLE-12-SP2-Server-POOL-x86_64-Build1415-Media1
REPO_8 	SLE-12-SP2-SDK-POOL-x86_64-Build0626-Media1
REPO_9 	SLE-12-SP2-SDK-POOL-x86_64-Build0626-Media1.license
SCC_EMAIL 	openqa@suse.de
SCC_REGCODE 	SLE-12-SP2-Server-DVD-x86_64-5UH8qgmU6TGmH1nP
SCC_URL 	http://scc.openqa.suse.de
SHUTDOWN_NEEDS_AUTH 	1
SLENKINS_TESTSUITES_REPO 	http://download.suse.de/ibs/Devel:/SLEnkins:/testsuites/SLE_12_SP2/
TEST 	BOOT_SNAP_NEW
UPGRADE 	1
VERSION 	12-SP2
VNC 	91

**Comment by sysrich** [comment]
You know..the reason I asked the questions about the variables was so you
could correct it BEFORE it broke in production ;)

It would be really helpful if you kept note of the variables you use in
testing, and included them in PR's for new scenarios

specifically we need just the settings we need in the scenario, not the
whole vars/params.json (which takes us a lot of diffing to find the minimal
changeset needed)

**Comment by MalloZup** [comment]
ok, i was thinking  that maybe we should make a TEMPLATE for PR in openqa
1) If (big_changes) then  Show a result/photo of the test.
2) if the you are creating a new test, show the vars.json vars

**Comment by sysrich** [comment]
Not a terrible idea - why don't you follow that practice in all of your pull requests, and if it works we'll make it a standard practice for all?

**Comment by MalloZup** [comment]
@sysrich  i will try this of course. :) +1 i hope only that my swap brain will remember it , but of course i will try this out :)

**Pull request merged**